iina/MainWindowController.swift

```diff
@@ -2059,11 +2059,35 @@ class MainWindowController: PlayerWindowController {
     }
   }
 
+  /// Determine if the thumbnail preview can be shown above the progress bar in the on screen controller..
+  ///
+  /// Normally the OSC's thumbnail preview is shown above the time preview. This is the preferred location. However the
+  /// thumbnail preview extends beyond the frame of the OSC. If the OSC is near the top of the window this could result
+  /// in the thumbnail extending outside of the window resulting in clipping. This method checks if there is room for the
```

Comment:
I feel like the long-term solution here would be to pull the thumbnail into its own window.

Reply:
On the text field and user interaction, in Xcode the behavior attribute, which can be selectable, editable or none, is set to none. I think user interaction is already disabled? So the mouse click goes up the responder chain and is handled by `ControlBarView`.

If the thumbnail was it its own window how would what work in full screen mode?

And of course I ignored the question of whether IINA should increase the size of its sliders in the spirt of Apple's Big Sur changes. For this fix I tried to make the minimum changes required to restore what I think was the intended behavior. Part of the reason for limiting the fix is due to my lack of UI experience.

Reply:
Hmm, my understanding is that if the view does not have user interaction enabled it should just remove itself from hit testing, causing whatever's under it to get the event. Is that not happening?

Also, there are various window flags that allow windows to show up in fullscreen mode.

Reply:
I found a `UIView` [hitTest](https://developer.apple.com/documentation/uikit/uiview/1622469-hittest) method. Not sure what the equivalent AppKit method is. That method does take into account the state of user interactions:
> This method ignores view objects that are hidden, that have disabled user interactions, or have an alpha level less than 0.01.

However I have not found anything about the mouse event being delivered to the view underneath. The UIKit method refers to traversing the view higharchy. In the case at hand the superview is the OSC. This results in the `mouseDown` method of `ControlBarView` being called which initiates dragging of the OSC. That was as far as I traced the behavior as I realized the intention was that these OSC elements not overlap.

On converting the screenshot preview to use a `NSWindow`, although I am usually in favor of adding some cleanup/refactoring in this case I'd rather postpone that past the planned bug fix release. There are a lot of issues that it would be nice to fix in the bug fix release that still need PRs. For example issue #3863. I've tracked down what is going wrong. I was hoping to hear why the code is treating the OSC as a part of the title bar before I remove that. And of course I really need to get back to reviewing the plugin system as that is blocking the creation of PRs to avoid generating merge conflicts.

If you really want me to I'll see if I can switch screenshot previews to use a `NSWindow`.

Reply:
Found the AppKit [hitTest](https://developer.apple.com/documentation/appkit/nsview/1483364-hittest) method. Unlike the UIKit method the documentation for the AppKit method only mentions ignoring hidden views, nothing about user interaction.

Reply:
I'll ask around but I would be very surprised if views that have user interaction disabled do anything meaningful on clicks

Reply:
FYI, I used `hitTest` in the plugin overlay view: https://github.com/iina/iina/blob/c9dcd4061ca14e0f0eb0b41d9a1bba88daf85f92/iina/PluginOverlayView.swift#L17

Overriding this method and returning `nil` should disable all interactions.

Reply:
Yes. But no need to do that for this issue?

The root cause of the problem is the miss-alignment of UI components caused by Apple increasing the size of sliders in Big Sur. Everything works as intended once the alignment problems are corrected.

Reply:
Be sure to look at the screenshots in the issue. They show the OSC before the changes in this commit and after.